streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -112,7 +112,14 @@ public StreamTask(int id,
         }
 
         // initialize the task by initializing all its processor nodes in the topology
-        this.topology.init(this.processorContext);
+        for (ProcessorNode node : this.topology.processors()) {
+            this.currNode = node;
+            try {
+                node.init(this.processorContext);
```

Comment:
Then we could change the logic of ProcessorContextImpl.schedule() to remove the searching right?